### PR TITLE
[ORT Mobile] Some updates to iOS/Android build settings

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Some updates to iOS/Android build settings
- Enable code signing for iOS in build directly
- Make check binary size for Android minimal build optional
- Move Android test in build.py to a separated function
- Add build_java options to Android CI

**Motivation and Context**
- Some user may want to sign the iOS binary directly in build
- Check binary size may cause test failure for minimal build if some ops are included (technically it's not a failure, nut will cause user confusion)
- Add build_java can detect issues such as https://github.com/microsoft/onnxruntime/issues/4976

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -1060,19 +1060,25 @@ Android NNAPI Execution Provider can be built using building commands in [Androi
...
 
-   The onnxruntime library is not code signed, it may be required or desired to code sign the library for iOS devices. For more information, see [Code Signing](https://developer.apple.com/support/code-signing/).
+   If the development team ID which has a valid code signing certificate is provided, Xcode will code sign the onnxruntime library in the building process, otherwise, the onnxruntime will be built without code signing. It may be required or desired to code sign the library for iOS devices. For more information, see [Code Signing](https://developer.apple.com/support/code-signing/).
 
 #### Build Instructions
 Run one of the following build scripts from the ONNX Runtime repository root,
+##### Cross build for iOS simulator
+```
+/build.sh --config <Release|Debug|RelWithDebInfo|MinSizeRel> --use_xcode \
+           --ios --ios_sysroot iphonesimulator --osx_arch x86_64 --apple_deploy_target <minimal iOS version>
```
Comment (skottmckay) [inline BUILD.md:1070]: nit: ./build.sh for both commands

**Review by skottmckay — DISMISSED** [review]
:shipit:

**Review by gwang-msft — COMMENTED** [review]
Do we want to include this in 1.5?

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**